src/utils/lang-utils.ts

```diff
@@ -5,16 +5,21 @@ type LanguageCode = keyof typeof constants.ALL_LANGUAGES;
 export const getInitialLanguage = () => {
   const url_params = new URLSearchParams(window.location.search);
   const query_lang = url_params.get("lang");
-  const local_storage_language = JSON.parse(localStorage.getItem(constants.LANGUAGE_KEY) as string);
+  let local_storage_language = localStorage.getItem(constants.LANGUAGE_KEY) ?? '"EN"';
+  if(!(/^".*"$/.test(local_storage_language))){
+    localStorage.setItem(constants.LANGUAGE_KEY, JSON.stringify(local_storage_language));
+    local_storage_language = localStorage.getItem(constants.LANGUAGE_KEY) as string;
+  }
+  const parsed_local_storage_language = JSON.parse(local_storage_language);
```

Comment:
This update ensures backward compatibility. Previously, we stored languages in the format `"EN"`, which can't be parsed. To address this, we added a check to determine if the localStorage value is in the old format. If it is, we stringify it (`'"EN"'`) and replace the old value. Then, we read the newly stored value in localStorage.